enterprise/tasks.py

```diff
@@ -182,3 +207,133 @@ def send_sso_configured_email(
         )
         LOGGER.exception(message)
         raise exc
+
+
+@shared_task
+@set_code_owner_attribute
+def send_group_membership_invitation_notification(
+    enterprise_customer_uuid,
+    group_membership_uuid,
+    budget_expiration,
+    catalog_uuid
+):
```

Comment:
One way you could make this task much more efficient is to have it process a list of of membership uuids, rather than a single membership per task invocation.  That will help cut down a lot on the number of calls to the catalog service, and more importantly, on the number of calls to Braze.  The Braze client methods to create aliases and send messages already accept lists of recipients as input, so it should be pretty easy to adjust the logic you already have to work on a list of members.